### PR TITLE
Use Rust crypto implementation

### PR DESCRIPTION
**Pull request by hughns**
This isolates the Rust crypto based changes from https://github.com/element-hq/element-call/pull/2467

### CODE REVIEW

**Comment by hughns** [comment]
n.b. whilst we can review this now, we might want to wait until we merge it until we plan out how we will do https://github.com/element-hq/element-call/issues/1281.

**Comment by hughns** [comment]
As per the discussion today, we should proceed with reviewing and merging this PR independent of #1281.

**Review by robintown — APPROVED** [review]

**Review by fkwp — APPROVED** [review]
lgtm

**Pull request merged**
